pkgs/applications/graphics/loupe/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, lib
+, fetchgit
+, rustPlatform
+, meson
+, pkg-config
+, cmake
+, gtk4
+, itstool
+, desktop-file-utils
+, ninja
+, gnome-builder
+, wrapGAppsHook4
+, libadwaita
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "loupe";
+  version = "44.alpha";
+
+  src = fetchgit {
+    url = "https://gitlab.gnome.org/BrainBlasted/loupe.git";
+    rev = "aa781d386485d76a647a7b8968b91a08e521c780";
+    sha256 = "sha256-pkHmwnK9lBjpFQbGkEjgTi5rOsYLMK6OydB/k6aMk84=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-ay2HNTcYumkwpUZkntJK4vA7puXz5swb8NwC8LG8vW8=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    cmake
+    gtk4
+    itstool
+    desktop-file-utils # update-desktop-database
+    ninja
+    gnome-builder
+    wrapGAppsHook4
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://gitlab.gnome.org/BrainBlasted/loupe/";
+    description = "A simple image viewer application written with GTK4 and Rust";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ jk ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
Does it not work on other platforms?

Reply:
Probably does but there's not much docs and that's the only platform I've tried so far

Reply:
Most of the other GTK3/4 applications in nixpkgs have platforms.unix here.